### PR TITLE
Consolidate github workflows

### PR DESCRIPTION
**Pull request by I-Al-Istannen**
This PR consolidates the basic workflows into one with multiple jobs. This is a bit nicer to look at and doesn't spam as much if things fail.

### CODE REVIEW

File: .github/workflows/basic-checks.yml [hunk 1]
```diff
@@ -1,9 +1,35 @@
-name: SonarCloud
+name: Basic checks
 
-on: [push, pull_request_target]
+on: [pull_request_target]
 
 jobs:
-  build:
+  autostlye:
```
Comment (marko-radosavljevic) [inline .github/workflows/basic-checks.yml:6]:
```suggestion
  spotless:
```

File: .github/workflows/basic-checks.yml [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+        with:
+          java-version: ${{ secrets.JAVA_VERSION }}
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - name: Check
+        run: ./gradlew spotlessCheck
+
+  tests:
+    name: "tests"
```
Comment (marko-radosavljevic) [inline .github/workflows/basic-checks.yml:21]:
```suggestion
    name: "Test"
```

File: .github/workflows/basic-checks.yml [hunk 3]
```diff
@@ -1,9 +1,35 @@
-name: SonarCloud
+name: Basic checks
```
Comment (marko-radosavljevic) [inline .github/workflows/basic-checks.yml:1]: Just basic? :weary:

Comment (I-Al-Istannen) [reply .github/workflows/basic-checks.yml:1]: Yea, they don't do too much :P

Comment (marko-radosavljevic) [reply .github/workflows/basic-checks.yml:1]: :hankey:

**Review by marko-radosavljevic — APPROVED** [review]

**Pull request merged**